File: pjt_tournois_e_sport_boot/src/main/java/projet/sopra/pjt_tournois_e_sport_boot/model/Championnat.java
package projet.sopra.pjt_tournois_e_sport_boot.model;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.persistence.Transient;

@Entity
@Table(name = "championnat")
public class Championnat extends Tournoi {

	/// ATTRIBUTES
	@OneToMany(mappedBy = "tournoi")
	private List<Journee> JourneesAJouerPoules = new ArrayList<Journee>();
	@OneToMany(mappedBy = "id")
	private List<Journee> JourneesAJouerFinales = new ArrayList<Journee>();
	@Transient
	private List<Poule> poules = new ArrayList<Poule>();
	@Column(name = "prochaine_etape_finale")
	private Etape prochaineEtape;

//	private int nbPoules = 0; remplace par prochaineEtape.getNbMatches();

	/// CONSTRUCTORS

	public Championnat() {

	}

	public Championnat(String nom, LocalDate dateDeDebut, String jeu, Set<Inscription> listeInscriptions) {
		super(nom, dateDeDebut, jeu, listeInscriptions);

	}

	/// GETTERS

	public List<Journee> getJourneesAJouerPoules() {
		return JourneesAJouerPoules;
	}

	public List<Journee> getJourneesAJouerFinales() {
		return JourneesAJouerFinales;
	}

	public List<Poule> getPoules() {
		return poules;
	}

	public Etape getProchaineEtape() {
		return prochaineEtape;
	}

//	public int getNbPoules() {
//		return nbPoules;
//	}
	/// SETTERS

	public void setJourneesAJouerPoules(List<Journee> journeesAJouerPoules) {
		JourneesAJouerPoules = journeesAJouerPoules;
	}

	public void setJourneesAJouerFinales(List<Journee> journeesAJouerFinales) {
		JourneesAJouerFinales = journeesAJouerFinales;
	}

	public void setPoules(List<Poule> poules) {
		this.poules = poules;
	}

	public void setProchaineEtape(Etape prochaineEtape) {
		this.prochaineEtape = prochaineEtape;
	}

//	public void setNbPoules(int nbPoules) {
//		this.nbPoules = nbPoules;
//	}

	
	

}

File: pjt_tournois_e_sport_angular/src/app/model/championnat.ts
import { Etape } from './etape.enum';
import { Journee } from './journee';
import { Poule } from './poule';
import { Tournoi } from './tournoi';

export class Championnat extends Tournoi {
  private _journeesAJouerPoules?: Journee[] | undefined;
  private _journeesAJouerFinales?: Journee[] | undefined;
  private _poules?: Poule[] | undefined;
  private _prochaineEtape?: Etape | undefined;

  constructor(
    journeesAJouerPoules?: Journee[],
    journeesAJouerFinales?: Journee[],
    poules?: Poule[],
    prochaineEtape?: Etape
  ) {
    super();

    this._journeesAJouerPoules = journeesAJouerPoules;
    this._journeesAJouerFinales = journeesAJouerFinales;
    this._poules = poules;
    this._prochaineEtape = prochaineEtape;
  }

  /**
   * Getter journeesAJouerPoules
   * @return {Journee[]}
   */

  public get journeesAJouerPoules(): Journee[] | undefined {
    return this._journeesAJouerPoules;
  }

  /**
   * Getter journeesAJouerFinales
   * @return {Journee[]}
   */

  public get journeesAJouerFinales(): Journee[] | undefined {
    return this._journeesAJouerFinales;
  }

  /**
   * Getter poules
   * @return {Poule[]}
   */

  public get poules(): Poule[] | undefined {
    return this._poules;
  }

  /**
   * Getter prochaineEtape
   * @return {Etape}
   */

  public get prochaineEtape(): Etape | undefined {
    return this._prochaineEtape;
  }

  /**
   * Setter journeesAJouerPoules
   * @param {Journee[]} value
   */
  public set journeesAJouerPoules(value: Journee[] | undefined) {
    this._journeesAJouerPoules = value;
  }

  /**
   * Setter journeesAJouerFinales
   * @param {Journee[]} value
   */
  public set journeesAJouerFinales(value: Journee[] | undefined) {
    this._journeesAJouerFinales = value;
  }

  /**
   * Setter poules
   * @param {Poule[]} value
   */
  public set poules(value: Poule[] | undefined) {
    this._poules = value;
  }

  /**
   * Setter prochaineEtape
   * @param {Etape} value
   */
  public set prochaineEtape(value: Etape | undefined) {
    this._prochaineEtape = value;
  }
}
